util/fibers/detail/scheduler.cc

```diff
@@ -320,14 +320,18 @@ bool Scheduler::WaitUntil(chrono::steady_clock::time_point tp, FiberInterface* m
 bool Scheduler::ProcessRemoteReady(FiberInterface* active) {
   bool res = false;
   while (true) {
-    FiberInterface* fi = remote_ready_queue_.Pop();
-    if (!fi)
-      break;
+    auto [fi, qempty] = remote_ready_queue_.PopWeak();
+    if (!fi) {
+      if (qempty)
+        break;
+      DVLOG(1) << "Retrying ";
+      continue;
+    }
```

Comment:
Why do you use PopWeak here if you already fixed Pop to have retries? For the DVLOG?

Reply:
yes, also wanted to see how many times we iterate per attempt